Tests/test_content.py

```diff
@@ -1016,14 +1034,218 @@ def handle_github_response(response):
+        test_details: Details of the currently executed test
+        test_timeout: The time out
+        thread_index:
+
+    Returns:
+
+    """
+    locked_integrations = []
+    bucket = storage_client.bucket(BUCKET_NAME)
+    prints_manager.add_print_job(f'attempting to lock integrations: {integrations_generation_number.keys()}',
+                                 print,
+                                 thread_index,
+                                 include_timestamp=True)
+    for integration, generation_number in integrations_generation_number.items():
+        blob = bucket.blob(f'{LOCKS_PATH}/{integration}')
+        try:
+            blob.upload_from_string(f'{CIRCLE_BUILD_NUM}:{test_timeout + 30}', if_generation_match=generation_number)
+            prints_manager.add_print_job(f'integration {integration} locked',
+                                         print,
+                                         thread_index,
+                                         include_timestamp=True)
+            locked_integrations.append(integration)
+        except PreconditionFailed:
+            # if this exception occurs it means that another build has locked this integration
+            # before this build managed to do it.
+            # we need to unlock all the integrations we have already locked and try again later
+            prints_manager.add_print_job(
+                f'Could not lock integration {integration}, delaying test execution for test with details: {test_details}',
+                print_warning,
+                thread_index,
+                include_timestamp=True)
+            unlock_integrations(locked_integrations, prints_manager, storage_client, thread_index)
+            return False
+    return True
+
+
+def unlock_integrations(locked_integrations: list,
+                        prints_manager: ParallelPrintsManager,
+                        storage_client: storage.Client,
+                        thread_index: int) -> None:
+    """
+    Delete all integration lock files for integrations specified in 'locked_integrations'
+    Args:
+        locked_integrations: Integration to unlock
+        prints_manager: ParallelPrintsManager object
+        storage_client: The GCP storage client
+        thread_index: The index of the thread that executes the unlocking
+    """
+    locked_integration_blobs = get_locked_integrations(locked_integrations, storage_client)
+    for integration, lock_file in locked_integration_blobs.items():
+        try:
+            lock_file.delete(if_generation_match=lock_file.generation)
+        except PreconditionFailed:
+            prints_manager.add_print_job(f'Could not unlock integration {integration}',
+                                         print_warning,
+                                         thread_index,
+                                         include_timestamp=True)
+
+
+def get_locked_integrations(integrations: list, storage_client: storage.Client) -> dict:
```

Comment:
You are basically getting the details about locks on a subset of integrations 
```suggestion
def get_locked_details(integrations: list, storage_client: storage.Client) -> dict:
```

Reply:
Not exactly- I'm getting the blob objects that represent the lock file **only** for the integrations that have a lock file.

So the question that is asked in this method is more `given a list of integrations- which of them are locked?`

Potentially if non are locked this could be an empty dict.